app/services/foreman/plugin.rb

```diff
@@ -234,15 +238,34 @@ def permission(name, hash, options = {})
 
     # Add a new role if it doesn't exist
     def role(name, permissions)
-      @default_roles[name] = permissions
+      default_roles[name] = permissions
       return false if pending_migrations || Rails.env.test?
 
       Role.transaction do
         role = Role.where(:name => name).first_or_create
         role.add_permissions!(permissions) if role.permissions.empty?
+        rbac_registry.role_ids << role.id
       end
     end
 
+    # Add plugin permissions to core's Manager and Viewer roles
+    # Usage:
+    # add_resource_permissions_to_default_roles ['MyPlugin::FirstResource', 'MyPlugin::SecondResource'], :except => [:skip_this_permission]
+    def add_resource_permissions_to_default_roles(resources, opts = {})
+      Plugin::RbacSupport.new.add_resource_permissions_to_default_roles resources, opts
+    end
+
+    # Add plugin permissions to Manager and Viewer roles. Use this for permissions without resource_type or to handle special cases
+    # Usage:
+    # add_permissions_to_default_roles 'Role Name' => [:first_permission, :second_permission]
+    def add_permissions_to_default_roles(args)
+      Plugin::RbacSupport.new.add_permissions_to_default_roles args
+    end
+
+    def add_all_permissions_to_default_roles
```

Comment:
Thanks, would you mind adding the similar docs as for add_permissions_to_default_roles (mentioning it adds all registered permissions to Manager and Viewer roles). Also a test for this method would be useful I think.